lib/gutentag/tagged_with_query.rb

```diff
@@ -29,8 +29,9 @@ def id_query
   end
 
   def name_query
+    tag_names = tags.map { |t| Gutentag.normaliser.call(t.is_a?(Gutentag::Tag) ? t.name : t) }
```

Comment:
Let's pull this line out into a private method (and switch the `t` variable to be `tag`)